crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -429,10 +429,17 @@ impl Instruction {
                         // Delete the constrain, it is always true
                         return Ok(NodeEval::VarOrInstruction(NodeId::dummy()));
                     } else if obj.is_zero() {
-                        return Err(RuntimeErrorKind::UnstructuredError {
-                            message: "Constraint is always false".into(),
+                        if let Some(location) = *location {
+                            return Err(RuntimeErrorKind::UnstructuredError {
+                                message: "Constraint is always false".into(),
+                            }
+                            .add_location(location));
+                        } else {
+                            return Err(RuntimeErrorKind::Spanless(
+                                "Constraint is always false".into(),
+                            )
+                            .add_location(Location::dummy()));
```

Comment:
Is there anything better we can do here? Reporting "constraint is always false" with whichever location in the compiler happens to be assigned ID 0 doesn't seem too helpful. I think reporting with no Location at all would be less confusing than reporting with an incorrect location here.

In general we should avoid dummy values whenever possible, they are an anti-pattern like null. `Option<T>` should be preferred. Perhaps we need a `.no_location()` option to convert RuntimeErrorKinds into RuntimeErrors without a location. Then we'd need support for reporting these as well.

Reply:
I agree, I did not want to impact on the error system, but now I suggest the following:
- I create an issue for having block locations
- I update this PR to use optional locations in runtime error and constrains.

Note also that with the assert!() feature, we would be able to add better message to the short-circuit constraints (e.g related to the out-of-bounds)